pep-0681.rst

```diff
@@ -366,6 +366,11 @@ descriptor classes typically use common parameter names to construct
 these field descriptors. This specification formalizes the names and
 meanings of the parameters that must be understood for static type
 checkers. These standardized parameters must be keyword-only.
+
+These parameters are a superset of those supported by
```

Comment:
@JelleZijlstra, our hope is to keep `dataclass_transform` behavior (ex. field descriptor parameters) in sync with stdlib `dataclass` (as appropriate) so future new behavior in `dataclass` could also be used in `dataclass_transform`. That's good for library authors and type checker maintainers. But should the PEP express that desire? The implications of any future `dataclass` enhancement on `dataclass_transform` would need to be thought though, so implying that new `dataclasses.field` parameters should automatically be supported by `dataclass_transform` field descriptors doesn't seem safe.

Are there other PEPs/features that are related in this way?

Reply:
I think it would be good to say that explicitly in the PEP. I can't think of a similar situation elsewhere in Python; maybe the closest analog is that when Python itself adds a new feature (e.g., match statements, exception groups), type checkers add support for it too.

It would be good though to check in with @ericvsmith (the maintainer of dataclasses in the stdlib).

Reply:
I'm not sure exactly what the question is. But I do think I should have worked more closely with the typing community before adding kw_only, and I think in general new or changed python features should take typing into account.

Reply:
No specific question, just making sure you're aware of this PEP and the impact on dataclasses itself.

Reply:
@JelleZijlstra, I added a paragraph in the Rationale section about adding future `dataclass` features to `dataclass_transform`